LTD-5963 Add security classification screen in product/approval details flow

markj0hnst0n opened the pull request:
### Aim

Security classification is needed specifically in the product wizard.  A confirmation screen is added first so that this details screen appears conditionally.

[LTD-5963](https://uktrade.atlassian.net/browse/LTD-5963)


[LTD-5963]: https://uktrade.atlassian.net/browse/LTD-5963?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

currycoder commented on exporter/f680/application_sections/approval_details/forms.py:
```diff
@@ -130,6 +131,82 @@ def get_layout_fields(self):
+    )
+    security_classification = forms.ChoiceField(
+        choices=(
+            Choice("unclassified", "Unclassified"),
+            Choice("official", "Official"),
+            Choice("official-sensitive", "Official-Sensitive"),
+            Choice("restricted", "Restricted"),
+            Choice("confidential", "Confidential"),
+            Choice("secret", "Secret"),
+            Choice("top-secret", "Top Secret", divider="Or"),
+            Choice("other", "Other"),
```
Can these be married up with the gradings here https://github.com/uktrade/lite-frontend/blob/dev/exporter/f680/application_sections/user_information/forms.py#L145-L155

I know that we want distinct lists, but where the strings overlap I believe that we should use shared constants.

currycoder replied:
Not sure if I've caught this mid-update but would be good if this followed the same pattern as the security release choices

markj0hnst0n commented on exporter/f680/application_sections/approval_details/views.py:
```diff
@@ -53,6 +62,7 @@ class ProductInformationView(F680ApplicationSectionWizard):
     condition_dict = {
         FormSteps.PRODUCT_CONTROLLED_UNDER_ITAR: is_foreign_tech_or_information_shared,
         FormSteps.PRODUCT_CONTROLLED_UNDER_ITAR_DETAILS: is_controlled_under_itar,
+        FormSteps.PRODUCT_SECURITY_CLASSIFICATION_DETAILS: has_security_classification,
```
Got some feedback from Karwai.  They can apply before it is security classified but they need to provide more information about what steps they have taken to get it classified.  She has updated the screens on the mural board and i've added a conditional radio button accordingly

markj0hnst0n commented on exporter/f680/application_sections/approval_details/forms.py:
```diff
@@ -13,6 +14,7 @@
     ConditionalRadiosQuestion,
 )
 from core.forms.utils import coerce_str_to_bool
+from ...constants import SecurityGrading
```
No worries.  Changed now.

depsiatwal commented on exporter/f680/application_sections/approval_details/views.py:
```diff
@@ -36,10 +39,23 @@ def is_controlled_under_itar(wizard):
+        (FormSteps.PRODUCT_SECURITY_CLASSIFICATION_DETAILS, ProductSecurityClassificationForm),
+        (FormSteps.ACTION_TAKEN_TO_CLASSIFY_PRODUCT, ActionTakenToClassifyInfo),
         (FormSteps.PRODUCT_FOREIGN_TECHNOLOGY_OR_INFORMATION_SHARED, ProductForeignTechOrSharedInformation),
         (FormSteps.PRODUCT_CONTROLLED_UNDER_ITAR, ProductControlledUnderItar),
```
Also when we select the product is under ITAR controlled there's no validation to ensure we fill in the text field.

markj0hnst0n commented:
@depsiatwal I've addressed the comments regarding required conditional textboxes for 'product is under ITAR controlled',  'Does the item include cryptography or other information security features?' and I've also added this for 'Will the item be used by the UK Armed Forces?'.  Date input validation has also been addressed.

depsiatwal reviewed (approved):
lgtm
